torchtext/data/functional.py

```diff
@@ -182,5 +181,54 @@ def numericalize_tokens_from_iterator(vocab, iterator, removed_tokens=None):
         if removed_tokens is None:
             yield iter(vocab[token] for token in tokens)
         else:
-            yield iter(map(lambda x: vocab[x],
-                       filter(lambda x: x not in removed_tokens, tokens)))
+            yield iter(
+                map(lambda x: vocab[x],
+                    filter(lambda x: x not in removed_tokens, tokens)))
+
+
+def read_text_iterator(path, tokenizer):
+    r"""Read text from path and yield a list of tokens based on the tokenizer
+    Arguments:
+        path: the file path.
+        tokenizer: the tokenizer used to tokenize string text.
+    Examples:
+        >>> from torchtext.data.functional import read_text_iterator
+        >>> tokenizer = get_tokenizer("basic_english")
+        >>> list((read_text_iterator('.data/ptb.train.txt', tokenizer)))
+            [['Sentencepiece', 'encode', 'as', 'pieces'], ['example', 'to', 'try!']]
+    """
+
+    with io.open(path, encoding="utf8") as f:
+        reader = unicode_csv_reader(f)
+        for row in reader:
+            tokens = tokenizer(' '.join(row))
+            yield tokens
```

Comment:
Since torchscript doesn't support generators, I'd not add those iterators here (once we add something, it's hard to remove).